numpy/core/src/common/simd/avx2/arithmetic.h

```diff
@@ -127,6 +135,19 @@ NPY_FINLINE npy_uint32 npyv_sum_u32(__m256i a)
     return _mm_cvtsi128_si32(s1);
 }
 
+NPY_FINLINE npy_uint32 npyv_sum_u16(__m256i a)
+{
+    npyv_u32x2 res = npyv_expand_u32_u16(a);
+    return (unsigned)npyv_sum_u32(_mm256_add_epi32(res.val[0], res.val[1]));
+}
+
+NPY_FINLINE npy_uint64 npyv_sum_u64(__m256i a)
+{
+    npy_uint64 NPY_DECL_ALIGNED(32) idx[2];
+    _mm_store_si128((__m128i*)idx, _mm_add_epi64(_mm256_castsi256_si128(a), _mm256_extracti128_si256(a, 1)));
+    return idx[0] + idx[1];
+}
+
```

Comment:
```suggestion

NPY_FINLINE npy_uint64 npyv_sum_u64(npyv_u64 a)
{
    __m256i two = _mm256_add_epi64(a, _mm256_shuffle_epi32(a, _MM_SHUFFLE(1, 0, 3, 2)));
    __m128i one = _mm_add_epi64(_mm256_castsi256_si128(two), _mm256_extracti128_si256(two, 1));
    return (npy_uint64)npyv128_cvtsi128_si64(one);
}
```
shuffle + add + split + add + movq should performs better than fall-back to scalar addition through 2 movq.
NOTE: `movq`  a.k.a `_mm_cvtsi128_si64` isn't available in 32-bit mode, so you will have to emulate it.
You should define `npy_int64 npyv128_cvtsi128_si64(__m128i)` inside a new utils header for the `SSE` branch to share with AVX2 & AVX512F similar to the AVX2 one.

Reply:
The straightforward solution to simulate int64 extract action in 32-bit mode is to use slow `_mm_store_si128`,  but I'm afraid it will have a bad performance impact.

Reply:
It's fine, the compiler going to optimize it via `movd + suffle + movd`.